src/components/ProfileEditModal/ProfileEditModal.tsx

```diff
@@ -1,33 +1,79 @@
-import React, { useState } from "react";
+import React, { useState, useContext, useEffect } from "react";
 import styles from "./styles";
 import { Modal, View, Pressable, TextInput, Text } from "react-native";
 import SvgIcon from "../../../assets/SvgIcon";
 import globalStyles from "../../globalStyles";
+import { User } from "../../types/schema";
+import userContext from "../../context/userContext";
+import firebaseApp from "../../firebase/firebaseApp";
+import { getUser } from "../../firebase/firestore/user";
+
+export type UserUpdateData = {
+  full_name?: string;
+  role?: string;
+};
+
+type ProfileEditModalProps = {
+  visible: boolean;
+  setVisible: Function;
+  user: User;
+  setUser: any;
+};
+
+export const ProfileEditModal = ({
+  visible,
+  setVisible,
+  user,
+  setUser,
+}: ProfileEditModalProps) => {
+  const db = firebaseApp.firestore().collection("users");
```

Comment:
we shouldn't have any firebase content outside of the firebase folder! see below comment